Allow compression to be disabled

hatchan opened the pull request:
The wrangler environment already takes care of this, so if we do it in our app then we potentially compress it twice, and thus the client would only see a compressed payload and not the JSON payload

arendjr reviewed (approved):
Very nice! I'm liking this custom hooks library more and more :)

hatchan committed:
Allow compression to be disabled

The wrangler environment already takes care of this, so if we do it in our app then we potentially compress it twice, and thus the client would only see a compressed payload and not the JSON payload

stephlow reviewed (approved):
Nice! Thanks for the quick fix

hatchan committed:
Resolve clippies
